packages/concerto-core/lib/introspect/validator.js

```diff
@@ -46,10 +47,11 @@ class Validator {
     /**
      * @param {string} id the identifier of the instance
      * @param {string} msg the exception message
+     * @param {string} errorType the type of error
      * @throws {Error} throws an error to report the message
      */
-    reportError(id, msg) {
-        throw new Error( 'Validator error for field `' + id + '`. ' + this.getFieldOrScalarDeclaration().getFullyQualifiedName() + ': ' + msg );
+    reportError(id, msg, errorType=ErrorCodes.DEFAULT_VALIDATOR_EXCEPTION) {
+        throw new BaseException('Validator error for field `' + id + '`. ' + this.getFieldOrScalarDeclaration().getFullyQualifiedName() + ': ' + msg ,errorType);
```

Comment:
removed StringValidatorExceptionClass as its not needed anymore I have added a optional param errorType in BaseException which solves the purpose